fs/operations/operations.go

```diff
@@ -830,59 +830,144 @@ func ListLong(ctx context.Context, f fs.Fs, w io.Writer) error {
 // excludes
 //
 // Lists in parallel which may get them out of order
-func Md5sum(ctx context.Context, f fs.Fs, w io.Writer) error {
-	return HashLister(ctx, hash.MD5, f, w)
-}
+// func Md5sum(ctx context.Context, f fs.Fs, w io.Writer) error {
+// 	return HashLister(ctx, hash.MD5, false, false, f, w)
+// }
 
 // Sha1sum list the Fs to the supplied writer
 //
 // Obeys includes and excludes
 //
 // Lists in parallel which may get them out of order
-func Sha1sum(ctx context.Context, f fs.Fs, w io.Writer) error {
-	return HashLister(ctx, hash.SHA1, f, w)
-}
+// func Sha1sum(ctx context.Context, f fs.Fs, w io.Writer) error {
+// 	return HashLister(ctx, hash.SHA1, false, false, f, w)
+// }
 
 // hashSum returns the human readable hash for ht passed in.  This may
 // be UNSUPPORTED or ERROR. If it isn't returning a valid hash it will
 // return an error.
-func hashSum(ctx context.Context, ht hash.Type, o fs.Object) (string, error) {
+func hashSum(ctx context.Context, ht hash.Type, downloadFlag bool, o fs.Object) (string, error) {
+	var sum string
 	var err error
-	tr := accounting.Stats(ctx).NewCheckingTransfer(o)
-	defer func() {
-		tr.Done(ctx, err)
-	}()
-	sum, err := o.Hash(ctx, ht)
-	if err == hash.ErrUnsupported {
-		sum = "UNSUPPORTED"
-	} else if err != nil {
-		fs.Debugf(o, "Failed to read %v: %v", ht, err)
-		sum = "ERROR"
+
+	// If downloadFlag is true, download and hash the file.
+	// If downloadFlag is false, call o.Hash asking the backend for the hash
+	if downloadFlag {
+		
+		// Setup accounting as NewTransfer.  Same approach used in Cat
```

Comment:
Flagging questions as I realized this has the rough comments I wrote when coding it and not the slightly refined versions I posted to the forum.

Here: confirm account usage is correct

Reply:
Yes `NewTransfer` is correct